Allow use of remote source if URI is specified as database.

tmertens opened the pull request:
Enable pushing to / pulling from a remote postgres server when a full connection url is passed as the local database parameter instead of just a database name.

This was required in order to perform automated deploys from a Dockerized jenkins instance with a linked postgres Docker container where the containers are running on the same docker host, but the database connection must occur via the linked container's virtual IP and the database is not accessible on the jenkins server via `localhost`.

Josh-Burgess commented:
:+1:

fledman commented:
:+1:

uhoh-itsmaciek commented on lib/heroku/command/pg.rb:
```diff
@@ -458,6 +452,14 @@ def generate_resolver
 
+  def parse_db_uri(local)
+    if local =~ %r(://)
+      return "postgres:///#{local}"
```
Isn't this logic backwards?

tmertens replied:
Whoops, yeah it looks like it.  I will fix and add a test.

dickeyxxx commented:
this looks useful, we may need to update the docs a bit to include it

tmertens commented:
@uhoh-itsmaciek Thanks for the catch, it is fixed now and I added tests for the db string parser.
@dickeyxxx I updated the code documentation in pg.rb to reflect the new behavior.

dickeyxxx commented:
:+1:
